src/bootloaders/systemd-class.c

```diff
@@ -241,12 +241,17 @@ bool sd_class_install_kernel(const BootManager *manager, const Kernel *kernel)
                                  get_kernel_destination_impl(manager),
                                  kernel->target.path);
         /* Optional initrd */
+        if (!boot_manager_write_initrd_freestanding(manager, writer)) {
```

Comment:
Should we also handle the syslinux case? @ikeydoherty I'm not sure how much you care about how this or the grub portion works though, comments?

Reply:
This does introduce a risk btw, if an initrd used is updated but has an issue there won't be a fallback initrd that can be used to load the system creating a single point of failure (the system might no longer boot in cases such as an encrypted rootfs for example). I'm not sure what mitigration we could add though. @ikeydoherty @busykai thoughts?

Reply:
I can implement for grub and syslinux, but those are not in clear, it makes me slower testing it

Reply:
added